compiler/jsgen.nim

```diff
@@ -790,19 +790,34 @@ proc needsNoCopy(y: PNode): bool =
 proc genAsgnAux(p: PProc, x, y: PNode, noCopyNeeded: bool) =
   var a, b: TCompRes
   gen(p, x, a)
+
+  let xtyp = mapType(x.typ)
+
+  if x.kind == nkHiddenDeref and x.sons[0].kind == nkCall and xtyp != etyObject:
+    gen(p, x.sons[0], a)
+    addf(p.body, "nimVarUnpack = $1;$n", [a.rdLoc])
```

Comment:
there are some mechanisms to generate a unique identifier. I don't think you can just shadow whatever 'nimVarUnpack' variable might already exist.